sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -490,10 +599,31 @@ trait DivModLike extends BinaryArithmetic {
       s"${eval2.value} == 0"
     }
     val javaType = CodeGenerator.javaType(dataType)
-    val operation = if (operandsDataType.isInstanceOf[DecimalType]) {
-      decimalToDataTypeCodeGen(s"${eval1.value}.$decimalMethod(${eval2.value})")
+    val (checkOverflow, operation) = if (operandsDataType.isInstanceOf[DecimalType]) {
+      val decimal = super.dataType.asInstanceOf[DecimalType]
+      val errorContextCode = if (nullOnOverflow) {
+        "\"\""
+      } else {
+        ctx.addReferenceObj("errCtx", queryContext)
+      }
+      val decimalType = CodeGenerator.javaType(decimal)
+      val decimalValue = ctx.freshName("decimalValue")
+      val operationValue = ctx.freshName("operationValue")
+      // scalastyle:off line.size.limit
+      (s"""
+          |$decimalType $decimalValue = ${eval1.value}.$decimalMethod(${eval2.value}).toPrecision(
```

Comment:
The final code is `${ev.value} = $operation;`, I don't think the code here can compile.